core/src/main/scala/kafka/server/BrokerLifecycleManager.scala

```diff
@@ -166,6 +166,19 @@ class BrokerLifecycleManager(
    */
   private var registered = false
 
+  /**
+   * True if a request has been sent and a response or timeout has not yet been processed.
```

Comment:
a request => a Heartbeat request ?

Reply:
It applies to both Heartbeat and Registration requests which are the only requests this manager sends.

I think we're also fixing a bug were a second registration request could have been scheduled after a call to `propagateDirectoryFailure` and before the registration response is received.

Reply:
Hmm, if that's the case, should we set communicationInFlight to true when sending the very first Registration request?

Reply:
We do. The very first Registration request is also scheduled via a `CommunicationEvent`. 
`communicationInFlight = true` is now set after both `_channelManager.sendRequest(new BrokerRegistrationRequest...` and `_channelManager.sendRequest(new BrokerHeartbeatRequest...`